src/algebra/category/Module/abelian.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import algebra.category.Module.kernels
+import algebra.category.Module.limits
+import category_theory.abelian.basic
+
+/-!
+# The category of left R-modules is abelian.
+-/
+
+open category_theory
+open category_theory.limits
+
+noncomputable theory
+
+universe u
+
+namespace Module
+variables {R : Type u} [ring R] {M N : Module R} (f : M ⟶ N)
+
+/-- In the category of modules, every monomorphism is normal. -/
+def normal_mono [mono f] : normal_mono f :=
+{ Z := of R f.range.quotient,
+  g := f.range.mkq,
+  w := linear_map.range_mkq_comp _,
+  is_limit :=
+    is_kernel.iso_kernel _ _ (kernel_is_limit _)
+      (linear_equiv.to_Module_iso'
+        (linear_equiv.trans (submodule.quot_equiv_of_eq_bot _ (ker_eq_bot_of_mono _)).symm
+          (linear_equiv.trans (linear_map.quot_ker_equiv_range f)
+            (linear_equiv.of_eq _ _ (submodule.ker_mkq _).symm)))) $
```

Comment:
I wish we could use `calc` blocks with these sort of equivalence relations. It would be so much easier to follow this!

Reply:
Agreed!

Reply:
I've added some comments which hopefully demystify that part a bit.